src/Nfield.Quota/QuotaVariableDefinition.cs

```diff
@@ -21,5 +22,63 @@ public QuotaVariableDefinition(IEnumerable<QuotaLevelDefinition> levels)
         public string OdinVariableName { get; set; }
 
         public ICollection<QuotaLevelDefinition> Levels { get; }
+        
+        public static bool operator ==(QuotaVariableDefinition existingDefinition, QuotaVariableDefinition newDefinition)
+        {
+            return existingDefinition?.Equals(newDefinition) ?? false;
+        }
+
+        public static bool operator !=(QuotaVariableDefinition existingDefinition, QuotaVariableDefinition newDefinition)
+        {
+            return !(existingDefinition == newDefinition);
+        }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as QuotaVariableDefinition;
+            return other != null && Equals(other);
+        }
+
+        public override int GetHashCode()
+        {
+            return Id.GetHashCode() ^ Name.GetHashCode() ^ OdinVariableName.GetHashCode() ^ Levels.GetHashCode();
+        }
+
+        public bool Equals(QuotaVariableDefinition other)
+        {
+            return other != null 
+                && (Id == other.Id) 
+                && (Name == other.Name) 
+                && (OdinVariableName == other.OdinVariableName) 
+                && ScrambledLevelsEquals(Levels,other.Levels);
+        }
+        private static bool ScrambledLevelsEquals<T>(IEnumerable<T> existingDefinition, IEnumerable<T> newDefinition)
+        {
+            var levels = new Dictionary<T, int>();
```

Comment:
I don't really understand the code below; I think it can be simpler:

```
existingDefinition.Count() == newDefinition.Count() && existingDefinition.All((item) => newDefinition.Contains(item));
```